source/extensions/filters/network/dubbo_proxy/router/router.h

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/router/router.h"
+
+#include "extensions/filters/network/dubbo_proxy/metadata.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace DubboProxy {
+namespace Router {
+
+/**
+ * RouteEntry is an individual resolved route entry.
+ */
+class RouteEntry {
+public:
+  virtual ~RouteEntry() {}
+
+  /**
+   * @return const std::string& the upstream cluster that owns the route.
+   */
+  virtual const std::string& clusterName() const PURE;
+
+  /**
+   * @return MetadataMatchCriteria* the metadata that a subset load balancer should match when
+   * selecting an upstream host
+   */
+  virtual const Envoy::Router::MetadataMatchCriteria* metadataMatchCriteria() const PURE;
```

Comment:
`Envoy::Router::MetadataMatchCriteria`   forward statement?

Reply:
I don't think it is appropriate to use the forward statement here. the router interface is provided to the user, and I want to express the dependency explicitly. If this is a implement class, it is best to use the forward statement.